packages/sdk/src/evm/contracts/smart-contract.ts

```diff
@@ -335,6 +338,10 @@ export class SmartContract<
     return assertEnabled(this.detectAccount(), FEATURE_ACCOUNT);
   }
 
+  get baseRouter(): BaseRouterClass<BaseRouter> {
```

Comment:
lets call this `extensions` 

and make the api
`contract.extensions.add("DirectListingLogic")`
`contract.extensions.update("DirectListingLogic")`
`contract.extensions.remove("DirectListingLogic")`

publisher as optional, default to deployer.thirdweb.eth